src/components/checkbox/checkbox.tsx

```diff
@@ -94,6 +96,40 @@ export class Checkbox {
     @Prop({ reflect: true })
     public required: boolean = false;
 
+    /**
+     * The label to show, when the component is `readonly` and its `value` is `true`.
+     * This can be used to clarify what kind of data is being visualized.
+     * If not set, the `label` property will be used.
+     */
+    @Prop({ reflect: true })
+    public readonlyTrueLabel?: string;
+
+    /**
+     * The label to show, when the component is `readonly` and its `value` is `false`.
+     * This can be used to clarify what kind of data is being visualized.
+     * If not set, the `label` property will be used.
+     */
+    @Prop({ reflect: true })
+    public readonlyFalseLabel?: string;
+
+    /**
+     * The icon to show, when the component is `readonly` and its `value` is `true`.
+     * This can be used to clarify what kind of data is being visualized.
+     * If not set, a default icon with a default color will be used.
+     * Colors can be customized using the `Icon` interface.
+     */
+    @Prop({ reflect: true })
+    public readonlyTrueIcon?: string | Icon;
+
+    /**
+     * The icon to show, when the component is `readonly` and its `value` is `false`.
+     * This can be used to clarify what kind of data is being visualized.
+     * If not set, a default icon with a default color will be used.
+     * Colors can be customized using the `Icon` interface.
+     */
+    @Prop({ reflect: true })
+    public readonlyFalseIcon?: string | Icon;
```

Comment:
I'm not super happy with the naming of these and the fact that there are four of them. Having four properties controlling just this single aspect of the component feels a bit weird.

I think we should combine them to a single property and give it a good name/interface. I can't think of anything right now, but I'm thinking something along the lines of this

```ts
type ReadOnly {
    label: string;
    icon: string | Icon;
}

type ReadOnlyProps = Record<boolean, ReadOnly>
```

Reply:
I agree. Let me see what I can do

Reply:
I did what I think you meant @jgroth can you check?